pilot/pkg/bootstrap/server.go

```diff
@@ -557,7 +558,7 @@ func (s *Server) initDNSTLSListener(dns string, tlsOptions TLSOptions) error {
 	cfg := &tls.Config{
 		GetCertificate: s.getIstiodCertificate,
 		ClientAuth:     tls.NoClientCert,
-		ClientCAs:      root,
+		ClientCAs:      certPool,
```

Comment:
If certPool is not nil but ClientAuth==tls.NoClientCert, does Istiod use the cert pool to verify client cert? Could you add a comment about the behavior?